araproc/analysis/standard_reco.py

```diff
@@ -924,3 +926,346 @@ def get_avg_corr_snr(self, wave_packet, excluded_channels = []):
+                delay = 0
+            else: 
+                
+                # Load  the cross correlation for this channel and the reference 
+                xcorr_times, xcorr_volts = wu.tgraph_to_arrays(
+                    self.__get_correlation_function(
+                        ch_ID, reference_ch, wavepacket, applyHilbert=False
+                    )
+                )
+
+                # Identify the `zoom_window` nanosecond window around the 
+                #   reconstructed delay
+                zoomed_indices = np.where(
+                    (np.abs( xcorr_times - reco_delays[ch_ID] )) < zoom_window // 2
+                )[0]
+
+                # Calculate the time of maximum correlation from this
+                #   window of expected signal delay.
+                if len(zoomed_indices) == 0: 
+
+                    # Software triggers are so short, a channel may not have
+                    #   signal during the time window where signal is expected.
+                    #   As a result, `zoom_indices` will have no entries. 
+                    # Just find the time of peak correlation for the full 
+                    #   waveform in this scenario.
+                    delay = xcorr_times[ np.nanargmax(xcorr_volts) ]
+
+                    # If the event can't find the time window to zoom in on
+                    #   and its not a software trigger, warn user
+                    if trig_type != "soft": 
+                        print(
+                            f"Touble calculating csw for {trig_type} event with "
+                            f"channels {ch_ID} and {reference_ch}")
+
+                else: 
+                    delay = xcorr_times[ 
+                        np.nanargmax(xcorr_volts[zoomed_indices]) # index of max xcorr in zoomed array
+                        + zoomed_indices[0] # Adjusted by first zoomed_index
+                    ]
+
+            # AraRoot always compares channels with smaller IDs to channels with 
+            #   larger IDs but we always want to compare to the reference channel.
+            # Correct for this if the current ch_ID is larger than the reference ch_ID
+            if ch_ID > reference_ch: 
+                delay *= -1
+            
+            # Save the calculated arrival delay
+            delays[ch_ID] = delay
+
+        return delays
+
+    def __trim_array(self, times, values, trim):
+        """
+        Trim an array by `trim` values, being careful to avoind trimming out the
+        maximum signal
+
+        Parameters
+        ----------
+        times : Iterable
+        values : Iterable
+        trim : int
+
+        Returns
+        -------
+        times : Iterable
+        values : Iterable
+        """
+
+        # Soft exit if the user doesn't actually request a trim
+        if trim == 0: 
+            return times, values
+        
+        # Identify the index where the peak signal is held
+        value_max_idx = np.nanargmax(values)
```

Comment:
Just `np.argmax`

Reply:
The scipy interpolator sometimes uses `nan` values when asked to extrapolate beyond the times it interpolated over. Usually I'd have 1 `nan` value per interpolation at the front or back of a waveform, I can switch to just trimming off the end of the waveforms if you'd prefer that

Reply:
Ahh right okay. Blerg that used to be configurable but it seems like its the only option now. Maybe the move is right after the interpolation to do a `yint = np.nan_to_num(yint)`. Then we don't have to deal with all these `np.nan*` functions?

Reply:
I actually just made a change that has the following logic: If we need to rebin the waveform, we know the first element will be `nan` due to this scipy extrapolation thing so we just trim that single element off of the waveform. I've now switched all the numpy functions with `nan` in the name to not have `nan` in the name. 

If you'd rather, I can tell the scipy interpolator to extrapolate to determine that first datapoint of the waveform.

Reply:
The way I've coded this, the first element will always be `nan` if we have to rebin the waveform. I always subtract off the "fraction of a bin shift" so the first element will always be for `original_times[0] - rebinning_shift` which is outside of the interpolation and always `nan`.

Making the waveform 0 I think could bias the RMS of the waveform so I'm favoring just deleting the first bin from the waveform.

Reply:
I actually just got rid of the whole coding block that introduces `nan`s and changed all the `np.nan*()` functions to not ignore nan values so this is resolved.